bluepysnap/circuit_validation.py

```diff
@@ -303,17 +303,18 @@ def _check_nodes_population(nodes_dict, config):
         if not nodes or len(nodes) == 0:
             errors.append(fatal('No "nodes" in {}.'.format(nodes_file)))
             return errors
-        if len(nodes.keys()) > 1:
-            required_datasets += ['node_group_id', 'node_group_index']
         for population_name in nodes:
             population = nodes[population_name]
+            groups = [population[name] for name in population
+                      if isinstance(population[name], h5py.Group)]
+            if len(groups) > 1:
+                required_datasets += ['node_group_id', 'node_group_index']
```

Comment:
I do more checks for the edges iirc. 
Here https://github.com/BlueBrain/snap/pull/79/files#diff-ab66ad956573cb0b56e8d613adbc9ec6R448-R475
maybe this is worth it to implement ? Or do you cover all the possible usecases allready ?

Reply:
Done, please verify